mineinabyss-features/src/main/kotlin/com/mineinabyss/armorstandlock/ArmorStandLockingListener.kt

```diff
@@ -0,0 +1,73 @@
+package com.mineinabyss.armorstandlock
+
+import com.mineinabyss.components.armorstandlock.LockArmorStand
+import com.mineinabyss.components.playerData
+import com.mineinabyss.geary.minecraft.access.toGeary
+import com.mineinabyss.geary.minecraft.access.toGearyOrNull
+import com.mineinabyss.geary.minecraft.store.encodeComponentsTo
+import com.mineinabyss.idofront.messaging.error
+import org.bukkit.Material
+import org.bukkit.Sound
+import org.bukkit.block.BlockFace
+import org.bukkit.entity.ArmorStand
+import org.bukkit.entity.EntityType
+import org.bukkit.entity.Player
+import org.bukkit.event.EventHandler
+import org.bukkit.event.Listener
+import org.bukkit.event.block.Action
+import org.bukkit.event.entity.EntityDamageByEntityEvent
+import org.bukkit.event.player.PlayerInteractAtEntityEvent
+import org.bukkit.event.player.PlayerInteractEvent
+import org.bukkit.inventory.EquipmentSlot
+
+class ArmorStandLockingListener : Listener {
+
+    @EventHandler
+    fun PlayerInteractEvent.onPlacingArmorStand() {
+        if (player.inventory.itemInMainHand.type != Material.ARMOR_STAND) return
+        if (action != Action.RIGHT_CLICK_BLOCK) return
+        if (hand != EquipmentSlot.HAND) return
+        if (blockFace == BlockFace.DOWN) return
+        isCancelled = true
+
+        // Wacky code to mimic vanilla placing
+        val newLoc =
+            if (blockFace == BlockFace.UP)
+                player.getTargetBlockExact(6)?.location!!.toCenterLocation().apply { y += 0.5 }
+            else
+                player.getLastTwoTargetBlocks(null, 6).first()?.location!!.toCenterLocation().apply { y -= 0.5 }
+
+        val armorStand = newLoc.world.spawnEntity(newLoc, EntityType.ARMOR_STAND) as ArmorStand
```

Comment:
Idofront provides `location.spawn<ArmorStand>()`, also since you are calling `armorStand` as a variable several times below, that's a good sign you should use `apply`